File: android/app/src/main/java/com/suminchoi/coachapp/design/components/RcZoneBadge.kt
package com.suminchoi.coachapp.design.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp
import com.suminchoi.coachapp.core.model.Zone
import com.suminchoi.coachapp.core.model.label
import com.suminchoi.coachapp.core.model.toZone
import com.suminchoi.coachapp.design.ZoneColors
import com.suminchoi.coachapp.design.RcTypography

fun zoneColor(zone: Zone) = when (zone) {
    Zone.RECOVERY  -> ZoneColors.recovery
    Zone.BASE      -> ZoneColors.base
    Zone.THRESHOLD -> ZoneColors.threshold
    Zone.INTERVAL  -> ZoneColors.interval
    Zone.REST      -> ZoneColors.rest
    Zone.LONG      -> ZoneColors.long
}

@Composable
fun RcZoneBadge(sessionType: String, modifier: Modifier = Modifier) {
    val zone = sessionType.toZone()
    val color = zoneColor(zone)
    Row(
        modifier = modifier
            .clip(RoundedCornerShape(5.dp))
            .background(color.copy(alpha = 0.22f))
            .padding(horizontal = 8.dp, vertical = 3.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(5.dp),
    ) {
        Box(
            modifier = Modifier
                .size(5.dp)
                .clip(CircleShape)
                .background(color),
        )
        Text(
            text = zone.label().uppercase(),
            style = RcTypography.labelSmall,
            color = color,
        )
    }
}

@Composable
fun RcZoneDot(sessionType: String, modifier: Modifier = Modifier) {
    val color = zoneColor(sessionType.toZone())
    Box(
        modifier = modifier
            .size(8.dp)
            .clip(CircleShape)
            .background(color),
        contentAlignment = Alignment.Center,
    ) {}
}

File: ios/CoachApp/CoachApp/Design/Components/RCZoneBadge.swift
import SwiftUI

struct RCZoneBadge: View {
    let zone: ZoneColor

    var body: some View {
        Text(zone.label)
            .font(.rcCaption(11))
            .fontWeight(.semibold)
            .foregroundStyle(zone.color)
            .padding(.horizontal, Spacing.sm)
            .padding(.vertical, Spacing.xs)
            .background(zone.color.opacity(0.12), in: Capsule())
    }
}

struct RCZoneDot: View {
    let zone: ZoneColor
    var size: CGFloat = 8

    var body: some View {
        Circle()
            .fill(zone.color)
            .frame(width: size, height: size)
    }
}

#Preview {
    HStack(spacing: 8) {
        ForEach([ZoneColor.recovery, .base, .threshold, .interval, .rest, .long], id: \.rawValue) { zone in
            RCZoneBadge(zone: zone)
        }
    }
    .padding()
}
